kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+        
+        // check for reconciler group rerouting
+        boolean groupRerouting = parameterService.getParameterValueAsBoolean(ProcurementCardRerouteDocumentsStep.class, REROUTE_PCDO_DOCUMENTS_IND_PARM_NM);
+        
+        Collection<ProcurementCardDocument> documents = null;
+        documents = retrieveUAProcurementCardDocumentsToRoute(KewApiConstants.ROUTE_HEADER_ENROUTE_CD);
+        
+        if ( LOG.isInfoEnabled() ) {
+            LOG.info("PCards to Reroute: " + new Integer(documents.size()).toString());
+        }
+        for (ProcurementCardDocument pcardDocument : documents) {
+        
+        	String pcardDocumentId = pcardDocument.getObjectId();
+            //get document route node
+            List<RouteNodeInstance> routeNodeInstances = pcardDocument.getDocumentHeader().getWorkflowDocument().getCurrentRouteNodeInstances();
+            String node = routeNodeInstances.get(0).getName();
+            //get Procurement Card Default information
+            ProcurementCardDefault procurementCardHolderDetail = getProcurementCardDefault(pcardDocument.getProcurementCardHolder().getTransactionCreditCardNumber());
+            boolean hasReconciler = hasReconciler(procurementCardHolderDetail);
+            if (hasReconciler) {   
+                //step one, reroute documents enroute to the error account Fiscal officer
+                if (node.equals("AccountFullEdit") && 
+                        pcardDocument.getTargetAccountingLine(0).getAccountNumber().equals(getErrorAccountNumber())) {
+                    pcardDocument.getTargetAccountingLine(0).setChartOfAccountsCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getChartOfAccountsCode()) ? procurementCardHolderDetail.getChartOfAccountsCode() : getErrorChartCode());
+                    pcardDocument.getTargetAccountingLine(0).setAccountNumber(StringUtils.isNotEmpty(procurementCardHolderDetail.getAccountNumber()) ? procurementCardHolderDetail.getAccountNumber() : getErrorAccountNumber());
+                    pcardDocument.getTargetAccountingLine(0).setFinancialObjectCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getFinancialObjectCode()) ? procurementCardHolderDetail.getFinancialObjectCode() : getDefaultObjectCode());
+                    pcardDocument.getTargetAccountingLine(0).setSubAccountNumber(procurementCardHolderDetail.getSubAccountNumber());
+                    pcardDocument.getTargetAccountingLine(0).setFinancialSubObjectCode(procurementCardHolderDetail.getFinancialSubObjectCode());
+                    if ( LOG.isInfoEnabled() ) {
+                        LOG.info("Rerouting doc #" + pcardDocumentId + " at AccountFullEdit for error account Fiscal Officer.");
+                    }                        
+                    requeueDocument(pcardDocument, pcardDocumentId, node, "HasReconciler"); 
+                    // pause between reroutes
+                    try {
+                        Thread.sleep(3000);
```

Comment:
See @Perrinek 's comment regarding the top-level 5 minute sleep.

Essentially, this is a legacy hack. Fixing the underlying threading issue is unlikely worth the effort, and if that effort were to be undertaken, it would be on its own ticket.